### PR TITLE
feat: add support for selecting folder when creating box

### PR DESCRIPTION
**Pull request by CompuIves**
image
image
image

### CODE REVIEW

File: packages/app/src/app/components/Create/CreateBox/CreateBoxForm.tsx [hunk 1]
```diff
@@ -14,11 +14,14 @@ import { useWorkspaceSubscription } from 'app/hooks/useWorkspaceSubscription';
...
 import { sandboxes } from '@codesandbox/common/lib/utils/url-generator/dashboard';
 import { docsUrl } from '@codesandbox/common/lib/utils/url-generator';
+import { PATHED_SANDBOXES_FOLDER_QUERY } from 'app/pages/Dashboard/queries';
+import { Query } from 'react-apollo';
 import { CreateParams } from '../utils/types';
 
 interface CreateBoxFormProps {
   type: 'sandbox' | 'devbox';
-  isDraft?: boolean;
+  collectionId: string | null;
```
Comment (alexnm) [inline packages/app/src/app/components/Create/CreateBox/CreateBoxForm.tsx:23]:
```suggestion
  collectionId: string | undefined;
```

File: packages/app/src/app/components/Create/CreateBox/CreateBoxForm.tsx [hunk 2]
```diff
@@ -38,6 +42,7 @@ export const CreateBoxForm: React.FC<CreateBoxFormProps> = ({
   const effects = useEffects();
   const nameInputRef = useRef<HTMLInputElement>(null);
   const { isPro } = useWorkspaceSubscription();
+  const isDraft = collectionId == null;
```
Comment (alexnm) [inline packages/app/src/app/components/Create/CreateBox/CreateBoxForm.tsx:45]:
```suggestion
  const isDraft = collectionId === undefined;
```

I'd keep consistency and stronger checks if this was defined as string | undefined in the top component

File: packages/app/src/app/components/Create/CreateBox/CreateBoxForm.tsx [hunk 3]
```diff
@@ -124,6 +129,37 @@ export const CreateBoxForm: React.FC<CreateBoxFormProps> = ({
           />
         </Stack>
 
+        <Stack direction="vertical" gap={1}>
+          <Text size={3} as="label">
+            Folder
+          </Text>
+
+          <Query
```
Comment (alexnm) [inline packages/app/src/app/components/Create/CreateBox/CreateBoxForm.tsx:137]: didn't know we have this :D

**Review by alexnm — APPROVED** [review]
I think this is useful for further clarifying the difference between drafts and folder sandboxes. I wonder if we should move the VM specs to the left column, as it's more like the starting specs based on the template, rather than a choice

File: packages/app/src/app/components/Create/CreateBox/CreateBoxForm.tsx (hunk 3 above)
Comment (CompuIves) [reply packages/app/src/app/components/Create/CreateBox/CreateBoxForm.tsx:137]: I found it in another component I remembered (the sandbox move modal) where we show all collections 😄

**Review by danilowoz — COMMENTED** [review]
👍 LGTM

**Pull request merged**